parquet/src/arrow/arrow_writer/mod.rs

```diff
@@ -244,16 +244,35 @@ impl<W: Write + Send> ArrowWriter<W> {
         self.writer.append_key_value_metadata(kv_metadata)
     }
 
+    /// Returns a reference to the underlying writer.
+    pub fn inner(&self) -> &W {
+        self.writer.inner()
+    }
+
+    /// Returns a mutable reference to the underlying writer.
+    ///
+    /// It is inadvisable to directly write to the underlying writer.
```

Comment:
This is copied from the docs on `std::io::BufWriter`, which is effectively what this is

Reply:
Maybe we can add some additional commentary about *why* it is inadvisable and maybe when it is ok (clearly there is some usecase otherwise we wouldn't have it) for example

```suggestion
    /// WARNING: directly writing to the underlying writer
    /// prior to closing this ArrowWriter can result in corrupted 
    /// parquet files. 
    ///
    /// This API can be used to flush the underlying writer
```

I think the actual usecase in this PR is more complicated (the async writer writes directly into the buffer) so I probably don't fully understand what is going on / the use case. Perhaps all the more reason to improve the documentation 🤔